.github/workflows/main.yml

```diff
@@ -0,0 +1,43 @@
+#main.yml
+
+name: Main
+on: pull_request
+
+jobs:
+  checks:
+    runs-on: macOS-latest
+    steps:
+
+      - name: Project Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Compute key for CI cache
+        run: ./.github/workflows/compute-ci-cache-key.sh key.txt
```

Comment:
What's the advantage of pulling this into a separate file? The logic doesn't look very complicated, and can just be done in yaml. The advantage of inlining it is that there is one less file to have to go find and open up to figure out what this script actually does.

That said, I'm not sure the cache is actually worth it, see my other comment.

Reply:
> The logic doesn't look very complicated, and can just be done in yaml
Although the statement around the complexity is true, I already had some bad times with multiline Shell scripts inlined in YAML, pretty much because it is quite easy to break YAML parsers 😞 

This is the reason why I try to wrap such scripts in a single Script file - therefore going with a single line - instead of inlining. But if you understand that caching is not a big deal at all, we can just get rid of this `step` and this script :)

Does that make sense to you?